### PR TITLE
feat(nx): nx migration with next 13

### PR DESCRIPTION
**Pull request by Leomotors**
## Demo

[https://dev.cugetreg.com](https://dev.cugetreg.com)

Approve this if you see no error in console and from usage.

Next 13 according to docs only have breaking changes with `next/image` and `next/link`

## Note
- Migration is done using `nx migrate` command

### CODE REVIEW

**Comment by Leomotors** [comment]
Update: DO NOT MERGE yet, test are failing

**Comment by bombnp** [comment]
it's weird that the tests are still failing even though u already upgraded package.json to use 15.x.x.

have you found anything else yet? @Leomotors

**Comment by Leomotors** [comment]
@bombnp Unit Test Failure is caused by Jest 27 -> 28, fixing

File: apps/cugetreg-web/src/common/components/CookieBanner/index.tsx
Comment (bombnp) [inline apps/cugetreg-web/src/common/components/CookieBanner/index.tsx:56]: what exactly does this do?

File: migrations.json
Comment (bombnp) [inline migrations.json:1]: do we need to commit this? not sure

**Review by bombnp — APPROVED** [review]

**Comment by bombnp** [comment]
wait for review from @saenyakorn since i don't know much about nx

File: apps/cugetreg-web/src/common/components/CookieBanner/index.tsx
Comment (Leomotors) [reply apps/cugetreg-web/src/common/components/CookieBanner/index.tsx:56]: New `next/link` no longer required children `<a>` and if you put that it will cause an error.
We use custom components which have `<a>` here so we need `legacyBehavior` to retain the previous behavior

https://nextjs.org/docs/api-reference/next/link

File: migrations.json
Comment (Leomotors) [reply migrations.json:1]: It looks like a lock file for nx migration.
In Brikl they commit it so I followed them.

Comment (bombnp) [reply migrations.json:1]: i thought nx migration is just changing the source code in the git already, so i wonder why we're committing this too. Do we have to run the migration again in the future?

Comment (Leomotors) [reply migrations.json:1]: 🤔🤔🤔 Good question, I don't know 😂. Let's wait for some nx expert to answer this.

File: nx.json
Comment (saenyakorn) [inline nx.json:13]: Why did you delete this na?

File: nx.json
Comment (saenyakorn) [inline nx.json:28]: I don't understand this for now. So, I'll read Nx docs more. But can you tell me what it is? if possible

File: nx.json
Comment (saenyakorn) [inline nx.json:53]: Also, here

File: migrations.json
Comment (saenyakorn) [reply migrations.json:1]: No Nx expert in Thinc.. Can you ask Brikl members?

**Review by saenyakorn — APPROVED** [review]
Other than that LGTM

**Comment by saenyakorn** [comment]
I just fixed CI here #429. This PR should have no action error after the PR's merged

File: nx.json
Comment (Leomotors) [reply nx.json:53]: These are all automatically done my `nx migrate`

**Pull request merged**
